front/src/modules/simulationResult/components/SpaceTimeChartWrapper/useTrackOccupancy.ts

```diff
@@ -162,35 +196,136 @@ const useTrackOccupancy = ({
+            }
+          : null;
+
+      const bodyForPaced =
+        pacedIds.length > 0
+          ? {
+              operational_point_id: opId,
+              infra_id: infraId,
+              paced_train_ids: pacedIds.map(extractEditoastIdFromPacedTrainId),
+            }
+          : null;
+
+      const [trainScheduleResp, pacedResp] = await Promise.all([
+        bodyForTrainSchedules
+          ? postTrainScheduleTrackOccupancy({ body: bodyForTrainSchedules })
+          : Promise.resolve(undefined),
+        bodyForPaced
+          ? postPacedTrainTrackOccupancy({ body: bodyForPaced })
+          : Promise.resolve(undefined),
+      ]);
+
+      const zones: MovableOccupancyZone[] = [];
+
+      if (trainScheduleResp?.data) {
+        for (const [trackId, occupations] of Object.entries(trainScheduleResp.data)) {
+          for (const occupation of occupations) {
+            const trainId = formatEditoastIdToTrainScheduleId(occupation.train_schedule_id);
+            const train = trainsCollection[trainId];
+            if (!train) continue;
+
+            zones.push(
+              getMovableOccupancyZone(
+                trackId,
+                trainId,
+                occupation,
+                train.spaceTimeCurves,
+                train.name,
+                train.departureTime
+              )
+            );
+          }
+        }
+      }
+
+      if (pacedResp?.data) {
+        for (const [trackId, occupations] of Object.entries(pacedResp.data)) {
+          for (const occupation of occupations) {
+            const pacedId = formatEditoastIdToPacedTrainId(occupation.paced_train_id);
+            const train = trainsCollection[pacedId];
+            if (!train) continue;
+
+            if (!('paced' in train)) {
+              continue;
+            }
+
+            const occurrenceId =
+              occupation.type === 'CreatedException'
+                ? formatPacedTrainIdToExceptionId(pacedId, occupation.exception_key)
+                : formatPacedTrainIdToIndexedOccurrenceId(pacedId, occupation.index ?? 0);
+
+            const exception =
+              occupation.type === 'CreatedException'
+                ? train.exceptions.find((e) => e.key === occupation.exception_key)
+                : train.exceptions.find((e) => e.occurrence_index === (occupation.index ?? 0));
+
+            const spaceTimeCurves =
+              train.exceptionProjections.get(occurrenceId)?.spaceTimeCurves ??
+              train.spaceTimeCurves;
+
+            let trainId: TrainId;
+            let trainName: string;
+            let startTime: Date;
+
+            if (occupation.type === 'CreatedException') {
+              if (!exception?.start_time?.value) continue;
+              trainId = occurrenceId;
+              trainName = exception.train_name?.value ?? `${train.name}/+`;
+              startTime = new Date(exception.start_time.value);
+            } else {
+              const i = occupation.index ?? 0;
```

Comment:
In this case, you are manipulating a base occurrence or a modified occurrence, so you know that your index is defined 
```suggestion
              const i = occupation.index;
```